contracts/contracts/strategies/balancer/BaseBalancerStrategy.sol

```diff
@@ -0,0 +1,323 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/**
+ * @title OETH Base Balancer Abstract Strategy
+ * @author Origin Protocol Inc
+ */
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC20, InitializableAbstractStrategy } from "../../utils/InitializableAbstractStrategy.sol";
+import { IBalancerVault } from "../../interfaces/balancer/IBalancerVault.sol";
+import { IRateProvider } from "../../interfaces/balancer/IRateProvider.sol";
+import { VaultReentrancyLib } from "./VaultReentrancyLib.sol";
+import { IOracle } from "../../interfaces/IOracle.sol";
+import { IVault } from "../../interfaces/IVault.sol";
+import { IWstETH } from "../../interfaces/IWstETH.sol";
+import { IERC4626 } from "../../../lib/openzeppelin/interfaces/IERC4626.sol";
+import { StableMath } from "../../utils/StableMath.sol";
+import "hardhat/console.sol";
+
+abstract contract BaseBalancerStrategy is InitializableAbstractStrategy {
+    using SafeERC20 for IERC20;
+    using StableMath for uint256;
+    IBalancerVault internal immutable balancerVault =
+        IBalancerVault(0xBA12222222228d8Ba445958a75a0704d566BF2C8);
+    address internal immutable stEth =
+        0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84;
+    address internal immutable wstEth =
+        0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0;
+    address internal immutable frxEth =
+        0x5E8422345238F34275888049021821E8E08CAa1f;
+    address internal immutable sfrxEth =
+        0xac3E018457B222d93114458476f3E3416Abbe38F;
+
+    address internal pTokenAddress;
```

Comment:
Any storage variable that can't be changed with an admin function should be immutable in my opinion. This saves over 2k in gas reading the data from storage.

Reply:
👍